ui/apps/akasha/src/extensions/inline-editor.tsx

```diff
@@ -0,0 +1,341 @@
+    (props.extensionData.embedEntry && embeddedPost.isLoading)
+  ) {
+    return <>{t('Loading Editor')}</>;
+  }
+
+  const handleFollow = () => {
+    if (entryData?.author.pubKey) {
+      followReq.mutate(entryData?.author.pubKey);
+    }
+  };
+
+  const handleUnfollow = () => {
+    if (entryData.author.pubKey) {
+      unfollowReq.mutate(entryData?.author.pubKey);
+    }
+  };
+
+  const getPostLabel = () => {
+    switch (action) {
+      case 'edit':
+        return t('Save Changes');
+      case 'post':
+      case 'embed':
+        return t('Publish');
+      case 'reply':
+        return t('Reply');
+    }
+  };
+
+  const getPlaceholderLabel = () => {
+    switch (action) {
+      case 'reply':
+        return `${t('Reply to')} ${entryAuthorName || ''}`;
+      default:
+        return t('Share your thoughts');
+    }
+  };
+
+  const handleReplyPlaceholderClick = () => {
+    analyticsActions.trackEvent({
+      category: AnalyticsCategories.POST,
+      action: 'Reply Placeholder Clicked',
+    });
+  };
+
+  const commentEditorUi = (
+    <CommentEditor
+      avatar={profileDataReq.data?.avatar}
+      ethAddress={loginQuery.data?.ethAddress}
+      postLabel={getPostLabel()}
+      placeholderLabel={getPlaceholderLabel()}
+      emojiPlaceholderLabel={t('Search')}
+      disablePublishLabel={t('Authenticating')}
+      disablePublish={disablePublishing}
+      onPublish={handleEntryPublish}
+      linkPreview={entryData?.linkPreview}
+      uploadedImages={entryData?.images}
+      getLinkPreview={getLinkPreview}
+      getMentions={handleMentionQueryChange}
+      getTags={handleTagQueryChange}
+      tags={tagSearch.data}
+      mentions={mentionSearch.data}
+      uploadRequest={uploadMediaToTextile}
+      embedEntryData={embedEntryData}
+      editorState={action !== 'reply' ? entryData?.slateContent : null}
+      onPlaceholderClick={action === 'reply' ? handleReplyPlaceholderClick : null}
+      isShown={!!props.extensionData.isShown}
+    />
+  );
+
+  return (
+    <>
+      {profileDataReq.status === 'error' && <Box margin="small">Error occured</Box>}
+      {embeddedPost.status === 'error' && (
+        <Box margin="small">Error loading embedded content..</Box>
+      )}
+      {editingPost.status === 'error' && <Box margin="small">Error loading post</Box>}
+
+      {(profileDataReq.status === 'loading' ||
+        embeddedPost.status === 'loading' ||
+        editingPost.status === 'loading') && (
+        <Box margin="small">
+          <EntryCardLoading />
+        </Box>
+      )}
+
+      {(!editingPost.isLoading || !embeddedPost.isLoading) &&
+        profileDataReq.status === 'success' &&
+        (action === 'reply' ? (
+          <Box margin="medium" style={{ position: 'relative' }}>
+            {commentEditorUi}
+          </Box>
+        ) : (
+          commentEditorUi
+        ))}
+
+      {publishComment.isLoading && publishComment.variables.postID === props.extensionData.entryId && (
```

Comment:
Is this the pending publish state of a comment? I'm not sure why it should be here, shouldn't be prepended to the list of comments?

Reply:
The publish comment mutation is happening inside inline editor and it depends on the state of `publishComment`. That's why I brought it here. Another way could be I could publish an event for this and it could stay prepended to the list of comments. What do you think ?

Reply:
You can have the same logic as for the post publishing where useMutationListener hook is used, no need for an extra event. The post publishing also have pending state

Reply:
That's a very clean approach. Ok, I will use that.